src/hooks/useDSHeading.tsx

```diff
@@ -0,0 +1,37 @@
+import Heading from "../components/Heading/Heading";
+
+/**
+ * DS internal helper hook to render a default `h2` heading element if the
+ * passed title is a string. Otherwise, it will return the title as is if it
+ * is a JSX element.
+ */
+function useDSHeading({
```

Comment:
@bigfishdesign13 this is a hook that I'm thinking can help create a default `h2` DS heading element (like the `List` component), a custom string title, or a "this custom already has a custom heading" (like the `NewsletterSignup` component) prop value.

Let me know what you think.

Reply:
Yes, I like having a reusable hook like this.  I left a few comments.  But I am also curious, do hook nanmes have to start with `use`?  Is that common practice or a requirement?  Iin my mind, I want something like this to start with `set`.